dashboard/app/models/census/state_cs_offering.rb

```diff
@@ -85,7 +99,20 @@ def self.infer_no(state_code)
     INFERRED_NO_EXCLUSION_LIST.exclude? state_code.upcase
   end
 
-  def self.construct_state_school_id(state_code, row_hash)
+  def self.construct_state_school_id(state_code, row_hash, school_year)
+    # Using V2 format.
+    if state_uses_format_v2(state_code, school_year)
+      # The V2 format requires either (district_id and school_id) OR nces_id.
+      if row_hash['nces_id']
+        return School.find_by(id: row_hash['NCES'])&.state_school_id
+      elsif row_hash['district_id'] && row_hash['school_id']
+        return School.construct_state_school_id(state_code, row_hash['district_id'], row_hash['school_id'])
```

Comment:
I suggest using `state_school_id` if it is explicitly provided in its own column in the spreadsheet as a backup when NCES ID is not available, but I don’t recommend constructing a state school ID from district and school ID.

I’ve seen considerable variation in how states provide `district_id` and `school_id` to us and how `state_school_id` is constructed from those values.  Sometimes leading zeroes are trimmed on one or both of those ids and that is done inconsistently between the data provided in the spreadsheet and the `state_school_id` in our `schools` table.  Also, sometimes the `school_id` includes the `district_id` in the spreadsheet (`WA-00123-00123021` for school 21 in district 123) but not in our `schools table.